engine/src/flutter/shell/platform/darwin/macos/framework/Source/FlutterSurface.mm

```diff
@@ -74,16 +80,26 @@ + (FlutterSurface*)fromFlutterMetalTexture:(const FlutterMetalTexture*)texture {
   return (__bridge FlutterSurface*)texture->user_data;
 }
 
-+ (IOSurfaceRef)createIOSurfaceWithSize:(CGSize)size {
-  unsigned pixelFormat = kCVPixelFormatType_32BGRA;
-  unsigned bytesPerElement = 4;
++ (IOSurfaceRef)createIOSurfaceWithSize:(CGSize)size pixelFormat:(MTLPixelFormat)pixelFormat {
+  unsigned pixelByteFormat;
+  unsigned bytesPerElement;
+  if (pixelFormat == MTLPixelFormatBGRA8Unorm) {
+    pixelByteFormat = kCVPixelFormatType_32BGRA;
+    bytesPerElement = 4;
+  } else if (pixelFormat == MTLPixelFormatBGRA10_XR) {
```

Comment:
Can we make BGR10_XR here?  Do we need alpha?

Reply:
We do. In general windows on macOS can be transparent, the surface may be overlaid over a NSVisualEffect view, or the surface might be a platform view overlay. 

In case where none of these is true however, this would indeed save quite a bit of memory.